python/data/dataloader/reader.py

```diff
@@ -44,14 +53,8 @@ def __len__(self):
 
     def __next__(self):
         batch_indices = next(self._sampler_iter)
-        data = []
-        label = []
-        for idx in batch_indices:
-            data.append(self.dataset[idx][0])
-            label.append(self.dataset[idx][1])
-        data = np.array(data)
-        label = np.array(label)
-        return data, label
+        data = self.fetcher.fetch(batch_indices)
```

Comment:
We can add two method to dataset
(1) self.dataset.create_data_batch(data)
(2) self.dataset.create_label_batch(label)
Then all dataset dependent things place at dataset, for example CocoDataset. How to create data and label is determin by yourself.